src/logger.ts

```diff
@@ -0,0 +1,80 @@
+enum LogLevel {
+  NONE = 0,
+  DEBUG = 10,
+  INFO = 20,
+  WARN = 30,
+  ERROR = 40,
+  FATAL = 50,
+}
+
+export function debug<T>(
+  msg: (T extends (...args: unknown[]) => unknown ? never : T) | (() => T),
+): string {
+  return write(msg, LogLevel.DEBUG);
+}
+
+export function info<T>(
+  msg: (T extends (...args: unknown[]) => unknown ? never : T) | (() => T),
+): string {
+  return write(msg, LogLevel.INFO);
+}
+
+export function warn<T>(
+  msg: (T extends (...args: unknown[]) => unknown ? never : T) | (() => T),
+): string {
+  return write(msg, LogLevel.WARN);
+}
+
+export function error<T>(
+  msg: (T extends (...args: unknown[]) => unknown ? never : T) | (() => T),
+): string {
+  return write(msg, LogLevel.ERROR);
+}
+
+export function fatal<T>(
+  msg: (T extends (...args: unknown[]) => unknown ? never : T) | (() => T),
+): string {
+  return write(msg, LogLevel.FATAL);
+}
+
+function write<T>(
+  msg: (T extends (...args: unknown[]) => unknown ? never : T) | (() => T),
+  level: LogLevel,
+): string {
+  let res: T | undefined;
+  let logMessage: string;
+
+  if (msg instanceof Function) {
+    res = msg();
+    logMessage = toString(res);
+  } else {
+    logMessage = toString(msg);
+  }
+
+  const formattedLog =
```

Comment:
I wonder if we can conditionally apply this formatting only if we are running in a hosted Slack Cloud context.. Maybe detecting for a [reserved AWS Lambda environment variable](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime) via [`Deno.env`](https://doc.deno.land/deno/stable/~/Deno.env)? Or maybe the deno library running on Lambda that does the dispatching injects an environment variable to make that explicit.

The reason I ask is because when I used this logger and ran locally, the output was quite hard to read.

Reply:
🤔 This is interesting, while I agree that it makes the stdout from `slack run` much more complex to read. 
I believe that since the `slack run` still executes the function in AWS, we wouldn't be able to generate activities as expected and someone calling `apps.activities.list` for the "(dev)" app wouldn't have the same result as the `slack deploy` version of the app.

I'm not sure if this matters, or if we're comfortable with that...